scene/2d/physics_body_2d.h

```diff
@@ -310,18 +310,28 @@ class CharacterBody2D : public PhysicsBody2D {
 		MOTION_MODE_GROUNDED,
 		MOTION_MODE_FREE,
 	};
+	enum MovingPlatformApplyVelocityOnLeave {
+		PLATFORM_VEL_ON_LEAVE_ALWAYS,
+		PLATFORM_VEL_ON_LEAVE_UPWARD_ONLY,
+		PLATFORM_VEL_ON_LEAVE_NEVER,
+	};
 	bool move_and_slide();
 
-	const Vector2 &get_linear_velocity() const;
-	void set_linear_velocity(const Vector2 &p_velocity);
+	const Vector2 &get_motion_velocity() const;
+	void set_motion_velocity(const Vector2 &p_velocity);
 
 	bool is_on_floor() const;
 	bool is_on_floor_only() const;
 	bool is_on_wall() const;
 	bool is_on_wall_only() const;
 	bool is_on_ceiling() const;
 	bool is_on_ceiling_only() const;
+	Vector2 get_last_motion() const;
+	Vector2 get_position_delta() const;
 	Vector2 get_floor_normal() const;
+	Vector2 get_wall_normal() const;
+	Vector2 get_real_velocity() const;
+
 	real_t get_floor_angle(const Vector2 &p_up_direction = Vector2(0.0, -1.0)) const;
 	Vector2 get_platform_velocity() const;
```

Comment:
The methods returning `Vector2` could return `const Vector2 &` instead to avoid making a copy (except for `get_position_delta()` which has to return a copy because it's doing some calculations).